nzedb/processing/Videos.php

```diff
@@ -84,15 +84,283 @@ protected function getSiteIDFromVideoID($siteColumn, $videoID)
+			}
+			$res = $this->getByTitleLikeQuery($title4, $type, $source);
+			if (isset($res['id'])) {
+				return $res['id'];
+			}
+		}
+		return false;
+	}
+
+	/**
+	 * Supplementary function for getByTitle that queries for exact match
+	 *
+	 * @param        $title
+	 * @param        $type
+	 * @param int    $source
+	 *
+	 * @return array|bool
+	 */
+	public function getByTitleQuery($title, $type, $source = 0)
+	{
+		$return = false;
+		if ($title) {
+			$return = $this->pdo->queryOneRow(
+				sprintf("
+					SELECT v.id
+					FROM videos v
+					LEFT JOIN videos_akas va ON v.id = va.videos_id
+					WHERE (v.title = %1\$s OR va.title = %1\$s)
+					AND v.type = %2\$d %3\$s",
+					$this->pdo->escapeString($title),
+					$type,
+					($source > 0 ? 'AND v.source = ' . $source : '')
+				)
+			);
+		}
+		return $return;
+	}
+
+	/**
+	 * Supplementary function for getByTitle that queries for a like match
+	 *
+	 * @param        $title
+	 * @param        $type
+	 * @param int    $source
+	 *
+	 * @return array|bool
+	 */
+	public function getByTitleLikeQuery($title, $type, $source = 0)
+	{
+		$return = false;
+
+		if ($title) {
+			$return = $this->pdo->queryOneRow(
+				sprintf("
+					SELECT v.id
+					FROM videos v
+					LEFT JOIN videos_akas va ON v.id = va.videos_id
+					WHERE (v.title %1\$s
+					OR va.title %1\$s)
+					AND type = %2\$d %3\$s",
+					$this->pdo->likeString(rtrim($title, '%'), false, false),
+					$type,
+					($source > 0 ? 'AND v.source = ' . $source : '')
+				)
+			);
+		}
+		return $return;
+	}
+
+	/**
+	 * Inserts aliases for videos
+	 *
+	 * @param       $videoId
+	 * @param array $aliasArr
+	 */
+	public function addAliases($videoId, $aliasArr = array())
+	{
+		if (!empty($aliasArr) && $videoId > 0) {
+			foreach ($aliasArr AS $key => $title) {
+				// Check if we have the AKA already
+				$check = $this->getAliases(0, $title);
+
+				if ($check === false) {
+					$this->pdo->queryInsert(
+						sprintf('
+							INSERT INTO videos_akas
+							(videos_id, title)
+							VALUES (%d, %s)',
+							$videoId,
+							$this->pdo->escapeString($title)
+						)
+					);
+				}
+			}
+		}
+	}
+
+	/**
+	 * Retrieves all aliases for given VideoID or VideoID for a given alias
+	 *
+	 * @param int    $videoId
+	 * @param string $alias
+	 *
+	 * @return bool|\PDOStatement
+	 */
+	public function getAliases($videoId = 0, $alias = '')
+	{
+		$return = false;
+		$sql = '';
+
+		if ($videoId > 0) {
+			$sql = 'videos_id = ' . $videoId;
+		} else if ($alias !== '') {
+			$sql = 'title = ' . $this->pdo->escapeString($alias);
+		}
+
+		if ($sql !== '') {
+			$return = $this->pdo->query('
+				SELECT *
+				FROM videos_akas
+				WHERE ' . $sql, true
+			);
+		}
+		return (empty($return) ? false : $return);
+	}
+
+	/**
+	 * This function turns a roman numeral into an integer
+	 *
+	 * @param string $string
+	 *
+	 * @return int $e
+	 */
+	public function convertRomanToInt($string) {
```

Comment:
Function should be moved to Utils\Text probably. Although I think PHP has some support for using Roman Numerals.